icicle/tests/polynomial_test.cu

```diff
@@ -0,0 +1,915 @@
+
+#include <gtest/gtest.h>
+#include <iostream>
+#include <memory>
+#include <vector>
+#include <list>
+
+#include "curves/curve_config.cuh"
+using curve_config::affine_t;
+using curve_config::g2_affine_t;
+using curve_config::g2_projective_t;
+using curve_config::projective_t;
+using curve_config::scalar_t;
+
+#include "polynomials/polynomials.h"
+#include "polynomials/polynomials_c_api.h"
+#include "polynomials/cuda_backend/polynomial_cuda_backend.cuh"
+
+#include "appUtils/ntt/ntt.cuh"
+#include "appUtils/msm/msm.cuh"
+#include "utils/device_context.cuh"
+
+// using the MSM C-API directly since msm::MSM() symbol is hidden in icicle lib and I cannot understand why
```

Comment:
I mean the definition doesn't link. I did not want to include msm.cu since it complies slowly. I tried to explicitly instantiate MSM but it still doesn't link so I am using the C-API directly. Just for the test.